src/lib.rs

```diff
@@ -16,19 +20,53 @@ const CHUNK_SIZE: usize = 128; // Chunk size to be sent to Ledger
 
 /// Alias of `Vec<u8>`. The goal is naming to help understand what the bytes to deal with
 pub type BorshSerializedUnsignedTransaction = Vec<u8>;
+
+const SIGN_NORMAL: u8 = 0;
+const SIGN_NORMAL_LAST_CHUNK: u8 = 0x80;
+const SIGN_BLIND: u8 = 1;
+
+// this is value from LedgerHQ/app-near repo
+const SW_INCORRECT_P1_P2: u16 = 0x6A86;
+// this is value from LedgerHQ/app-near repo
+const SW_BUFFER_OVERFLOW: u16 = 0x6990;
+
+// this is value from LedgerHQ/app-near repo
+const SW_SETTING_BLIND_DISABLED: u16 = 0x6192;
+
 /// Alias of `Vec<u8>`. The goal is naming to help understand what the bytes to deal with
 pub type NEARLedgerAppVersion = Vec<u8>;
 /// Alias of `Vec<u8>`. The goal is naming to help understand what the bytes to deal with
 pub type SignatureBytes = Vec<u8>;
 
 #[derive(Debug)]
 pub enum NEARLedgerError {
-    /// Self-explanatory one
-    DeviceNotFound,
+    /// Error occuring on init of hidapid and getting current devices list
+    HidApiError(HidError),
+    /// Error occuring on creating a new hid transport, connecting to first ledger device found  
+    LedgerHidError(LedgerHIDError),
     /// Error occurred while exchanging with Ledger device
     APDUExchangeError(String),
+    /// Blind signature not supported
+    BlindSignatureNotSupported,
+    /// Blind signature disabled in ledger's app settings
+    BlindSignatureDisabled,
     /// Error with transport
     LedgerHIDError(LedgerHIDError),
+    /// Transaction is too large to be signed
+    BufferOverflow(OnlyBlindSigning),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct OnlyBlindSigning {
+    pub byte_array: CryptoHash,
+}
```

Comment:
How about turning it into?

```suggestion
    BufferOverflow { transaction_hash: CryptoHash },
}
```

given that we only pull in near-primitives-core for CryptoHash, we may even consider having our own `type CryptoHash = [u8; 32];`

Reply:
addressed by https://github.com/khorolets/near-ledger-rs/pull/4/commits/7729ef8bc0782688d2604f54f68109cfd0eb1865